src/main/java/io/leonis/zosma/ipc/peripheral/Controller.java

```diff
@@ -9,28 +9,30 @@
  * This interface describes the functionality of a controller which maps an entity's identifier
  * to the controls provided by the underlying implementation.
  *
- * @param <I> The type of identifier used to identify the controller
- * @param <S> The type of controls provided by the controller.
+ * @param <I> The type of identifier used to identify the controller.
  * @author Jeroen de Jong
  */
-public interface Controller<I extends Identity, S> extends Identity.Supplier {
+public interface Controller<I extends Identity> extends Identity.Supplier {
+
   /**
    * @return The identifier of the controller.
    */
   I getIdentity();
 
   /**
-   * @return The controls provided by the controller.
+   * @return The name of the controller.
    */
-  S getControls();
+  String getName();
 
   /**
-   * Supplies a mapping of {@link Controller} to {@link Identity} of objects it is controlling.
+   * Supplies a mapping of {@link Identity ControllerIdentity} to {@link Identity} of objects it is
+   * controlling.
    *
-   * @param <C> The type of {@link Controller} being supplied.
-   * @param <A> The type of {@link Identity} of the object it is controlling.
+   * @param <C> The type of {@link Identity ControllerIdentity}.
```

Comment:
you mean `the type of {@link Controller}`?

Reply:
no i mean that `C extends Controller` and you say here in the javadoc that it is `The type of {@link Identity ControllerIdentity}`

Reply:
Whilst it may be confusing; `C extends Identity` in the `MappingSupplier`.

https://github.com/delta-leonis/zosma/blob/90b2855c3e23f174e1784cacc9e2fa56726534b9/src/main/java/io/leonis/zosma/ipc/peripheral/Controller.java#L34